pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionMetaDataManager.java

```diff
@@ -290,26 +307,48 @@ public synchronized void giveupLeadership() {
      */
     public void processMetaDataTopicMessage(Message<byte[]> message) throws IOException {
         try {
-            Request.ServiceRequest serviceRequest = Request.ServiceRequest.parseFrom(message.getData());
-            if (log.isDebugEnabled()) {
-                log.debug("Received Service Request: {}", serviceRequest);
-            }
-            switch (serviceRequest.getServiceRequestType()) {
-                case UPDATE:
-                    this.processUpdate(serviceRequest.getFunctionMetaData());
-                    break;
-                case DELETE:
-                    this.proccessDeregister(serviceRequest.getFunctionMetaData());
-                    break;
-                default:
-                    log.warn("Received request with unrecognized type: {}", serviceRequest);
+            if (workerConfig.getCompactMetadataTopic()) {
```

Comment:
Or should we just check if the message message has a key or not?

Reply:
actually prefer explicit setting. 
Plus with this change, key is being set for whether you are compacting the topic or not

Reply:
Is there a way for a user to that is already using functions to be able to turn compaction on? Is there a path of migration for these users?

Reply:
The simplest thing maybe to change this check from

`if (workerConfig.getCompactMetadataTopic()` 

to

`if (messsage.hasKey())`

So old messages can get processed correctly as well as the the new messages in the new format

Reply:
The simplest thing maybe to change this check from

`if (workerConfig.getCompactMetadataTopic()` 

to

`if (messsage.getdata() == null)`

So old messages can get processed correctly as well as the the new messages in the new format

Reply:
The format of the messages are now different.

Reply:
Just the version is added as part of properties right?  We can still process the messages differently based on whether the data == null or not

Reply:
Not just the version. We actually write function metadata when compaction is turned on. In case when it’s not compacted, we write the service request for backwards compatibility.